[ONNX] Support exporting with Apex O2 (#65374)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66149 [ONNX] Update slice process shape to support rank only inference (#65782)
* **#66148 [ONNX] Support exporting with Apex O2 (#65374)**
* #66147 [ONNX] Symbolic: dynamic input for OneHot, bool for Einsum (#65940)
* #66146 [ONNX] Modify softplus symbolic to support beta!=1 (#65001)
* #66145 [ONNX] Deprecate fold_if pass (#65697)
* #66144 [ONNX] Add diagonal symbolic (#64454)
* #66143 ONNX: Delete or document skipped ORT tests (#64470)
* #66142 [ONNX] Add warning for inplace updates on tensor.shape in tracing mode (#63170)
* #66141 [ONNX] Update sum symbolic to handle dtypes (#64289)
* #66140 [ONNX] Export nn.Module call as ONNX local function (#63589)

Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D31424101](https://our.internmc.facebook.com/intern/diff/D31424101)

BowenBao committed:
[ONNX] Support exporting with Apex O2 (#65374)

Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support exporting with Apex O2 (#65374)"


Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31424101).

BowenBao committed:
Update on "[ONNX] Support exporting with Apex O2 (#65374)"


Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D31424101](https://our.internmc.facebook.com/intern/diff/D31424101)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support exporting with Apex O2 (#65374)"


Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D31424101](https://our.internmc.facebook.com/intern/diff/D31424101)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support exporting with Apex O2 (#65374)"


Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D31424101](https://our.internmc.facebook.com/intern/diff/D31424101)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support exporting with Apex O2 (#65374)"


Apex O2 hook state_dict to return fp16 weights as fp32. Exporter cannot identify them as same tensors.
Since this hook is only used by optimizer, it is safe to remove this hook while exporting.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D31424101](https://our.internmc.facebook.com/intern/diff/D31424101)

[ghstack-poisoned]
